ui/pages/confirmations/components/confirm/info/typed-sign/permit-simulation/permit-simulation.tsx

```diff
@@ -52,10 +53,15 @@ const PermitSimulation: React.FC<{
   }, [exchangeRate, value]);
 
   const { tokenValue, tokenValueMaxPrecision } = useMemo(() => {
-    const valueBN = new BigNumber(value / Math.pow(10, tokenDecimals));
```

Comment:
needed to split the BigNumber constants to use built in pow method since the result of `value / Math.pow(10, tokenDecimals)` could lead to an unexpected lengthy number. If it is over 15 digits it would cause the error.

We are also not applying to string on the result of the equation to pass to BigNumber directly because this could also result in precision loss.